docs/pages/[id].js

```diff
@@ -0,0 +1,65 @@
+// @flow strict
+import { MDXRemote } from 'next-mdx-remote';
+import { serialize } from 'next-mdx-remote/serialize';
+import { type Node } from 'react';
+
+import { getDocByRoute, getAllMarkdownPosts } from '../utils/mdHelper.js';
+import MarkdownPage from '../components/MarkdownPage.js';
+
+type MDXRemoteSerializeResult = {|
+  compiledSource: string,
+
+  frontmatter?: { [key: string]: string },
+|};
+
+type Props = {|
+  content: MDXRemoteSerializeResult,
+  meta: {|
+    title: string,
+    badge: 'pilot' | 'deprecated',
+    component?: boolean,
+    description: string,
+  |},
+  pageSourceUrl: string,
+|};
+
+export default function DocumentPage({ content, meta, pageSourceUrl }: Props): Node {
+  return (
+    <MarkdownPage meta={meta} pageSourceUrl={pageSourceUrl}>
+      <MDXRemote {...content} />
+    </MarkdownPage>
+  );
+}
+
+export async function getStaticProps(context: {| params: {| id: string |} |}): Promise<{|
+  props: {| meta: { [key: string]: string }, content: {||}, pageSourceUrl: string |},
+|}> {
+  const { id } = context.params;
+  const { meta, content } = getDocByRoute(id);
```

Comment:
same for getDocByRoute. HOw do you feel about moving function within getStaticProps or closer to where it's called.